From 716a6b5ecbb24bf7742c0da494db8c37893cd902 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Mon, 23 Apr 2018 11:17:23 +0300
Subject: [PATCH 3619/5242] MLK-18116: drm: imx: dcss: fix crtc enumeration
 problem

commit  ef14707fc38af37f4fd59cacb807bb89e94acd8a from
https://source.codeaurora.org/external/imx/linux-imx.git

The following commit:

44b460cfe554 ("drm: imx: remove struct imx_drm_crtc and
imx_drm_crtc_helper_funcs")

removed some functions from imx-drm-core. As a consequence,
the CRTC ports were not detected properly.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-crtc.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-crtc.c b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
index d02ab5a..64b1fe0 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-crtc.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
@@ -305,6 +305,7 @@ static int dcss_crtc_init(struct dcss_crtc *crtc,
 	if (IS_ERR(crtc->plane[0]))
 		return PTR_ERR(crtc->plane[0]);
 
+	crtc->base.port = pdata->of_node;
 	drm_crtc_helper_add(&crtc->base, &dcss_helper_funcs);
 	ret = drm_crtc_init_with_planes(drm, &crtc->base, &crtc->plane[0]->base,
 					NULL, &dcss_crtc_funcs, NULL);
-- 
1.7.9.5

